src/app-common/location-detail/Header.js

```diff
@@ -2,7 +2,12 @@ import React from "react";
 import PropTypes from 'prop-types';
 import "./location-detail.scss";
 import LocationStreamControls from "../../app-pages/map/components/location-stream-controls/LocationStreamControls"
-const LocationDetailHeader = ( { onExpand, locationDetail } ) => {
+const LocationDetailHeader = ( props ) => {
+  const {
+    onExpand,
+    /** @type a2w.models.LocationDetail */
```

Comment:
One thing I did notice is that the "type casting" only works if you pass in `props` to the component and then destructure them inside the function body. I think maybe the IDE gets confused between trying to use the JSDoc type and trying to use the type defined in propTypes.

Reply:
For some reason this doesn't work for me on vscode, even if I comment out the prop types. The `/** @type a2w.models.LocationSummary */` on the radar-time-series-bundle works though so I'm not sure what's going on. Either way we can merge it in

Reply:
I don't think you can destructure it. When you do const locationDetail = props.locationDetail, it works.

Reply:
Mohit when you say it doesn't work, do you mean you don't get the code completion? Or do you mean you get a build error and/or can't run the app? Just trying to make sure what you're seeing. We can discuss tomorrow. I'll wait to merge until I'm sure what you're seeing.